### DISCUSSION TITLE
Protocol improvement ideas for better apps & tools

### DISCUSSION THREAD

**Discussion by mishmosh** [filecoin-project/community#233] [Developers]
Here's a thread for discussing protocol improvements that would greatly benefit devtools builders (and, by extension, their users). Any themes that coalesce should be converted into feedback to the relevant Filecoin project teams or FIPs.

Initial topics:
* Testnet improvements (@therealjeffg)
* Tracking transactions (a wish from @jimmylee)

Thanks @autonome for kicking this off in Slack!

**Comment by mishmosh** [comment]
@therealjeffg What testnet are you using, calibration or nerpa? And what improvements would you like to see?

**Reply by jeffgca** [reply]
I think @walkah is working on that from our end so he would have the most detailed feedback. Essentially - we're forced to use mainnet in order to actually test things because estuary is using it, which means we need to toss real money around.

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
also nerpa is always behind. integrating nerpa into the network update flow would be great, at least when the mainnet migrates to a new version ALL testnets should be on that version too.

all miner side tooling will benefit from this since 512MB sectors lower the rtt for sector sealing to minutes

**Reply by jeffgca** [reply]
Having discussed this with the team, I think our issue currently is that testnet isn't suitable for the type of integration work we are doing.   This is because testnet does not accurately reflect mainnet (especially bids/deals), and mainnet has all the newest stuff deployed, so we are forced to test on in mainnet using real deals and balances. 

We would love to have a testnest that is functionally identical to mainnet but where we're doing pure simulations of deals and balances and still have confidence that what we're doing will work the same as on mainnet. We're not there yet.

**Reply by jimmylee** [reply]
I agree on establishing a testnet that is functionally identical to mainnet. This is holding back development work.
